src/main/java/org/sagebionetworks/web/server/servlet/filter/CrawlFilter.java

```diff
@@ -445,6 +479,72 @@ private String getEntityHtml(String entityId)
     return html.toString();
   }
 
+  public String getDatasetScriptElement(EntityBundle bundle)
+    throws JSONObjectAdapterException, RestServiceException {
+    StringBuilder html = new StringBuilder();
+    if (bundle.getEntity() instanceof Dataset) {
+      Dataset ds = (Dataset) bundle.getEntity();
+      html.append("<script type=\"application/ld+json\">");
+      JSONObjectAdapter json = new JSONObjectAdapterImpl();
+      json.put("@context", "http://schema.org/");
+      json.put("@type", "Dataset");
+      json.put("name", ds.getName());
+      json.put("description", ds.getDescription());
+      json.put(
+        "url",
+        "https://www.synapse.org/#!Synapse:" +
+        ds.getId() +
+        "." +
+        ds.getVersionNumber()
+      );
+      json.put("version", ds.getVersionNumber());
+
+      // add annotations
+      JSONArrayAdapter array = new JSONArrayAdapterImpl();
+      Map<String, AnnotationsValue> annotations = bundle
+        .getAnnotations()
+        .getAnnotations();
+      Set<String> annotationKeys = annotations.keySet();
+      int index = 0;
+      for (String key : annotationKeys) {
+        List<String> keyValuePairList = new ArrayList<String>();
+        keyValuePairList.add(key);
+        List<String> values = annotations.get(key).getValue();
+        for (String value : values) {
+          keyValuePairList.add(value);
+        }
+        array.put(index++, String.join(", ", keyValuePairList));
+      }
+      json.put("keywords", array);
+
+      // include identifier if there is a DOI association?
+
+      JSONObjectAdapter object = new JSONObjectAdapterImpl();
+      object.put("@type", "DataCatalog");
+      object.put("name", "Synapse");
+      object.put("url", "https://www.synapse.org");
+      json.put("includedInDataCatalog", object);
+
+      json.put("isAccessibleForFree", true);
+
+      json.put("dateModified", df.format(ds.getModifiedOn()));
+
+      object = new JSONObjectAdapterImpl();
+      object.put("@type", "Person");
+      UserProfile profile = synapseClient.getUserProfile(ds.getCreatedBy());
+      object.put("name", getDisplayName(profile));
+      object.put(
+        "url",
+        "https://www.synapse.org/#!Profile:" + ds.getCreatedBy()
+      );
+      json.put("creator", object);
```

Comment:
The Dataset owners can change the name and the description, but they have no control over this field.

Is there risk in surfacing datasets with the user who created it as an author when semantically it should be a consortia or different individual (e.g. the PI)? Proper attribution is critical to our mission, so I'd rather not have this than get this wrong, unless we either provide a way to override it, or to opt out of surfacing this document.